### PR TITLE
add new command SetManagerNic in redfish_config module

### LINKED ISSUE #64475: redfish_config - Manager - SetManagerNic
##### SUMMARY
This feature would implement a SetManagerNic command for the Manager category of redfish_config, to update Manager NIC configuration. 

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
redfish_config

##### ADDITIONAL INFORMATION
This command would help user to configure their Manager NIC.

### PR DESCRIPTION
**Pull request by panyy3**
##### SUMMARY
add SetManagerNic in redfish_config module to set Manager NIC configuration.

Fixes #64475 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
redfish_config

##### ADDITIONAL INFORMATION
none

### CODE REVIEW

File: lib/ansible/module_utils/redfish_utils.py [hunk 1]
```diff
@@ -2263,3 +2263,91 @@ def get_manager_health_report(self, manager_uri):
...
+        target_ethernet_current_setting = None
+        if nic_addr == 'null':
+            # Find root_uri matched EthernetInterface when nic_addr is not specified
+            nic_addr = (self.root_uri).split('/')[-1]
+        for uri in uris:
+            response = self.get_request(self.root_uri + uri)
+            if response['ret'] is False:
+                return response
+            data = response['data']
+            if nic_addr in str(data):
```
Comment (billdodd) [inline lib/ansible/module_utils/redfish_utils.py]: Let's say that `nic_addr` has a value of "192.168.0.1". And part of one of the `EthernetInterfaces` resource payloads (what is in the `data` var) looks like this:

```
    ...
    "IPv4Addresses": [
        {
            "Address": "192.168.0.10",
            "SubnetMask": "255.255.252.0",
            "AddressOrigin": "DHCP",
            "Gateway": "192.168.0.1",
            "Oem": {}
        }
    ],
    ...
```
If I'm interpreting this correctIy, I think the expression `if nic_addr in str(data)` is going to return True.
(e.g.: `if '192.168.0.1' in '... "Address": "192.168.0.10" ...'`)

Maybe we need a more explicit/rigorous way to select the EthernetInterface?

File: lib/ansible/module_utils/redfish_utils.py [hunk 2]
```diff
@@ -2263,3 +2263,91 @@ def get_manager_health_report(self, manager_uri):
...
+        data = response['data']
+        uris = [a.get('@odata.id') for a in data.get('Members', []) if
+                a.get('@odata.id')]
+
+        # Find target EthernetInterface
+        target_ethernet_uri = None
+        target_ethernet_current_setting = None
+        if nic_addr == 'null':
+            # Find root_uri matched EthernetInterface when nic_addr is not specified
+            nic_addr = (self.root_uri).split('/')[-1]
```
Comment (billdodd) [inline lib/ansible/module_utils/redfish_utils.py:2288]: The `root_uri` may include a port number (e.g. 192.168.1:8000). Should probably split that off also.

File: lib/ansible/module_utils/redfish_utils.py [hunk 3]
```diff
@@ -2263,3 +2263,91 @@ def get_manager_health_report(self, manager_uri):
...
+                else:
+                    return {'ret': False, 'msg': "Value of property %s in nic_config is invalid" % property}
+            else:
+                payload[property] = value
+
+        # If no need change, nothing to do. If error detected, report it
+        need_change = False
+        for property in payload.keys():
+            setValue = payload[property]
+            curValue = target_ethernet_current_setting[property]
```
Comment (billdodd) [inline lib/ansible/module_utils/redfish_utils.py]: nit: lower_case_with_underscores preferred over CamelCase per PEP8 (cur_value, set_value).

File: lib/ansible/module_utils/redfish_utils.py [hunk 4]
```diff
@@ -2263,3 +2263,91 @@ def get_manager_health_report(self, manager_uri):
...
+            if not isinstance(setValue, dict) and not isinstance(setValue, list):
+                if setValue != curValue:
+                    need_change = True
+            # type is dict
+            if isinstance(setValue, dict):
+                for subprop in payload[property].keys():
+                    if subprop not in target_ethernet_current_setting[property]:
+                        return {'ret': False, 'msg': "Sub-property %s in nic_config is invalid" % subprop}
+                    sub_setValue = payload[property][subprop]
+                    sub_curValue = target_ethernet_current_setting[property][subprop]
```
Comment (billdodd) [inline lib/ansible/module_utils/redfish_utils.py]: ditto

**Review by billdodd — COMMENTED** [review]
Looks pretty good!

I did have just a few comments. See below...

File: lib/ansible/module_utils/redfish_utils.py (hunk 1 above)
Comment (panyy3) [reply lib/ansible/module_utils/redfish_utils.py]: You are right. I have updated code to match '"192.168.0.1"' or "'192.168.0.1'" avoid partial match.

File: lib/ansible/module_utils/redfish_utils.py (hunk 2 above)
Comment (panyy3) [reply lib/ansible/module_utils/redfish_utils.py:2288]: Thanks for reminder, code is updated.

File: lib/ansible/module_utils/redfish_utils.py (hunk 3 above)
Comment (panyy3) [reply lib/ansible/module_utils/redfish_utils.py]: Got it, code is updated.

File: lib/ansible/module_utils/redfish_utils.py (hunk 4 above)
Comment (panyy3) [reply lib/ansible/module_utils/redfish_utils.py]: Got it, code is updated.

**Review by billdodd — COMMENTED** [review]
Thanks for the updates! LGTM

shipit

**Comment by mraineri** [comment]
shipit

**Comment by gundalow** [comment]
@panyy3 @mraineri @billdodd Thank you!

**Pull request merged**